lib/cylc/gui/dbchooser.py

```diff
@@ -202,52 +215,36 @@ def my_get_iter(item):
             iter = None
 
         # add new items at this level
-        for item in new_items:
+        for key in sorted(new_items):
+            if isinstance(key, tuple):
+                item = key[0]
+            else:
+                item = key
             if item not in old_items:
                 # new data wasn't in old - add it
-                if isinstance(new[item], dict):
+                if isinstance(new[key], dict):
                     xiter = ts.append(
                         piter, [item] + [None, None, None, None, None, None])
-                    self.build_treestore(new[item], xiter)
+                    self.build_treestore(new[key], xiter)
                 else:
-                    state, descr, dir = new[item]
+                    state, descr, dir = new[key]
                     yiter = ts.append(
-                        piter, [item] + new[item] + list(self.statecol(state)))
+                        piter, [item] + new[key] + list(self.statecol(state)))
             else:
                 # new data was already in old
-                if isinstance(new[item], dict):
+                if isinstance(new[key], dict):
                     # check lower levels
-                    niter = my_get_iter(item)
+                    niter = my_get_iter(key)
                     if niter:
                         chiter = ts.iter_children(niter)
                         if chiter:
-                            self.update_treestore(new[item], chiter)
+                            self.update_treestore(new[key], chiter)
 
     def run(self):
-        global debug
-        if debug:
-            print '* thread', self.me, 'starting'
+        """Main loop."""
         while not self.quit:
-            if self.running_choices_changed() or self.reload:
-                gobject.idle_add(self.update)
-            time.sleep(1)
-        else:
-            if debug:
-                print '* thread', self.me, 'quitting'
-            self.__class__.count -= 1
-
-    def running_choices_changed(self):
-        if not PyroInstalled:
-            return
-        # [(name, owner, host, port)]
-        results = scan(pyro_timeout=self.pyro_timeout)
-        choices = [(result[1]['name'], result[0]) for result in results]
-        choices.sort()
-        if choices != self.running_choices:
-            self.running_choices = choices
-            return True
-        else:
-            return False
+            gobject.idle_add(self.update)
+            sleep(0.1)
```

Comment:
I have found that 1 second was making the GUI a bit unresponsive, and could also have been the reason behind the strange blank screen + zombie behaviour. Reducing this to 0.1 makes the GUI more responsive, and I am no longer seeing the blank screen + zombie behaviour.

(See also the other logic, which delays calls to `scan_all` in 1-minute intervals.)